subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -519,8 +519,7 @@ static void recv_thread(void *p1, void *p2, void *p3)
 			 * unref on call to net_buf_frag_del
 			 */
 			frag = buf;
-			net_buf_ref(frag);
-			buf = net_buf_frag_del(NULL, frag);
+			buf = net_buf_frag_del(NULL, net_buf_ref(frag));
```

Comment:
Actually, I think the more correct thing would be to change `frag = buf;` into `frag = net_buf_unref(buf)`.

The reasoning: the code wants to store `frag` so that I can be later passed to `bt_recv()`. The `net_buf_frag_del()` line is just an iterator statement, and in fact the code should more correctly be passing `buf` as the input parameter (kind of like `i = i + 1;`) i.e. `buf = net_buf_frag_del(NULL, buf);`.

Also, please split the controller change into a separate commit. This is good practice in general, and it'll also give you a boost in contribution statistics 😉

Reply:
When you split into controller commit, do not add the `Fix` word in the commit message (i tend to relate `Fix` to associate with bug fix and needing backport if need be)